test/test_get_generated_files.py

```diff
@@ -0,0 +1,251 @@
+"""Test get_generated_files."""
+import pytest
+
+from castep_outputs.tools.get_generated_files import get_generated_files
+
+DUMMY_CELL = {
+    "positions_frac": {
+        ("Si", 1): {},
+        ("N", 1): {},
+    },
+}
+
+
+@pytest.mark.parametrize("seedname, expected", [
+    ("test", [
+        "test.bands",
+        "test.bib",
+        "test.castep",
+        "test.castep_bin",
+        "test.check",
+        "test.cst_esp",
+    ]),
+])
+def test_cli_like(seedname: str, expected: list[str]) -> None:
+    """Test via the CLI like interface parsing files on-the-fly."""
+    assert expected == get_generated_files(seedname)
+
+
+@pytest.mark.parametrize("cell, param, expected", [
+    (DUMMY_CELL,
+     {
+         "task": "moleculardynamics",
+         "metals_method": "dm",
+         "opt_strategy": "memory",
+         "xc_functional": "LDA",
+         "md_ensemble": "NVE",
+         "md_extrap": "mixed",
+     },
+     {
+         "seedname.*.drho2m",
+         "seedname.bib",
+         "seedname.cst_esp",
+         "seedname.*.drhom",
+         "seedname.castep",
+         "seedname.md",
+         "seedname.*.wf2m",
+         "seedname.castep_bin",
+         "seedname.*.wfm",
+         "seedname.bands",
+         "seedname.check",
+     }),
+
+    (DUMMY_CELL | {
+        "species_pot": {
+            "Si": {
+                "proj": "30N1.9:31N2.1:32N2.1",
+                "opt": [
+                    "tm",
+                    "nonlcc",
+                ],
+                "print": True,
+                "projectors": [
+                    {
+                        "orbital": 3,
+                        "shell": "s",
+                        "type": "N",
+                        "de": 1.9,
+                    },
+                    {
+                        "orbital": 3,
+                        "shell": "p",
+                        "type": "N",
+                        "de": 2.1,
+                    },
+                    {
+                        "orbital": 3,
+                        "shell": "d",
+                        "type": "N",
+                        "de": 2.1,
+                    },
+                ],
+                "string": "2|1.9|2.1|1.3|3.675|5.512|7.35|30N1.9:31N2.1:32N2.1(tm,nonlcc)[]",
+            },
+        },
+    },
+     {
+         "task": "singlepoint",
+         "metals_method": "dm",
+         "opt_strategy": "memory",
+         "xc_definition": {
+             "xc": {
+                 "LIBXC_GGA_XC_ZLP": 1.0,
```

Comment:
```suggestion
                 "LIBXC_GGA_XC_B97_D": 1.0,
```
ZLP is a meta-GGA, I guess you wanted to go GGA to get the PBE pseudopotential outputs!